src/models/feature_selection/shap_selector.py

```
@@ -0,0 +1,131 @@
+    try:
+        model = LGBMRegressor(n_estimators=500, learning_rate=0.05, n_jobs=-1)
+        model.fit(x_enc_train, y_train)
+    except Exception as e:
+        logger.error(f"LightGBM training failed: {e}")
+        raise
+
+    # 2) Prepare SHAP sample
+    n_val = x_enc_val.shape[0]
+    sample_n = min(shap_sample_size, n_val)
+    if sample_n <= 0:
+        raise ValueError("No validation rows available for SHAP computation")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)